src/ai.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "car.hpp"
+#include "direction.hpp"
+#include "lane.hpp"
+#include <cmath> 
+#include <cstdlib>
+#include <ctime>
+#include <limits>
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <vector>
+
+struct PlayerOutcome
+{
+	float cars_turned_per_move;
+	direction most_villainous_direction;
+};
+
+class AI
+{
+public:
+	const int MaxPly = 3;
+
+	// Estimated time it takes for the player to execute a move.
+	// Used to determine if the player must move a specific lane due to a time constraint
+	const float EstimatedPlayerMoveTime = 400;
```

Comment:
This isn't a super important constant but we can tweak it to get more desireable results.